service/storage_service.cc

```diff
@@ -1140,6 +1162,35 @@ class topology_coordinator {
                     break;
                 }
 
+                // Tell the node to shut down.
+                // This is done to improve user experience when there are no failures.
+                // In the next state (`node_state::left`), the node will be banned by the rest of the cluster,
+                // so there's no guarantee that it would learn about entering that state even if it was still
+                // a member of group0, hence we use a separate direct RPC in this state to shut it down.
+                //
+                // There is the possibility that the node will never get the message
+                // and decommission will hang on that node.
+                // This is fine for the rest of the cluster - we will still remove, ban the node and continue.
+                auto node_id = node.id;
+                bool shutdown_failed = false;
+                try {
+                    node = co_await exec_direct_command(std::move(node), raft_topology_cmd::command::shutdown);
+                } catch (...) {
```

Comment:
Shouldn't we catch and re-throw `term_changed_error`? OTOH `retake_node` below will do it anyway.